src/Domain/Donation.php

```diff
@@ -587,11 +589,53 @@ public function hasPostCreateUpdates(): bool
         return !in_array($this->getDonationStatus(), $this->newStatuses, true);
     }
 
+    /**
+     * @return string   The amount of the core donation, in £, which is to be paid out
+     *                  to the charity. This is the amount paid by the donor minus
+     *                  (a) any part of that amount which was a tip to the Big Give; and
+     *                  (b) fees on the remaining donation amount.
+     *                  It does not include separately sourced funds like matching or
+     *                  Gift Aid.
+     */
+    public function getAmountForCharity(): string
+    {
+        $feeRatio = '0.012';        // 1.2% of amount for charity (exc. tip)
+        $feeAmountFixed = '0.20';   // 20p fixed per-donation
+
+        // bcmath truncates values beyond the scale it's working at, so to get 1.2% and round
+        // in the normal mathematical way we need to start with 3 d.p. scale and round with a
+        // workaround.
+        $feeAmountFromPercentageComponent = $this->roundAmount(
+            bcmul($this->getAmount(), $feeRatio, 3)
+        );
+
+        return bcsub(
+            bcsub($this->getAmount(), $feeAmountFromPercentageComponent, 2),
+            $feeAmountFixed,
+            2
+        );
+    }
+
     /**
      * @return string[]
      */
     public static function getSuccessStatuses(): array
     {
         return self::$successStatuses;
     }
+
+    /**
+     * Takes a bcmath string amount with 3 or more decimal places and rounds to
+     * 2 places, with 0.005 rounding up and below rounding down.
+     *
+     * @param string $amount    Simplified from https://stackoverflow.com/a/51390451/2803757 for
+     *                          fixed scale and only positive inputs.
+     * @return string
+     */
+    private function roundAmount(string $amount): string
```

Comment:
@NoelLH Just for my own clarity, why are we passing an `amount` as a string and then rounding like this? Aren't there natural rounding php fns we can use?

Reply:
@tbg-jasonfung in your example, the type is a *float*. Floats should not be used for currencies in almost any language. [Some reasons why](https://spin.atomicobject.com/2014/08/14/currency-rounding-errors/).

PHP does not have a Decimal type, so the options are using integer pence values everywhere – sometimes good but sometimes confusing – or using bcmath. Until we had fee calculations here there were no multiplications on our side so we could always use 2-place precision consistently and not worry about this. Now we have to round to get the fees, using pence internally is a slightly more attractive option but I still don't think it's worth changing. This should be as complicated as it gets to keep using bcmath and we maintain all its benefits and the testing of amount handling logic we've done so far.